lib/colorls/flags.rb

```diff
@@ -37,12 +37,16 @@ def process
       # initialize locale from environment
       CLocale.setlocale(CLocale::LC_COLLATE, '')
 
-      return Core.new(@opts).ls if @args.empty?
+      @args = [Dir.pwd] if @args.empty?
```

Comment:
Is this required? I believe it is taken care of, in the `Core#new` method.

Reply:
This way there is only one code path and errors are handled in one place. Otherwise, I would have to enclose this line in a begin..rescue statement too.

Reply:
Maybe the handling of an "empty" input could be removed from `Core#new` then?

Reply:
Hmm, alright. Let's remove it there.

Reply:
OK, done.

I have also reduced the handled errors to subclasses of `SystemCallErrors` which are raised by calls to File.stat et cetera. This way other errors still abort the program, when something else breaks.